include/extractor/guidance/intersection_handler.hpp

```diff
@@ -327,6 +327,42 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
         return false;
     }();
 
+    // check whether we turn onto a oneway through street. These typically happen at the end of
+    // roads and might not seem obvious, since it isn't always as visible that you cannot turn
+    // left/right. To be on the safe side, we announce these as non-obvious
+    const auto turns_onto_through_street = [&](const typename IntersectionType::value_type &road) {
+        const auto in_through_candidate =
+            intersection.FindClosestBearing(util::bearing::reverse(road.bearing));
+
+        const auto &in_data = node_based_graph.GetEdgeData(in_through_candidate->eid);
+        const auto &out_data = node_based_graph.GetEdgeData(road.eid);
+
+        // by asking for the same class, we ensure that we do not overrule obvious by road-class
+        // decisions
+        const auto same_class = in_data.road_classification == out_data.road_classification;
```

Comment:
not necessarily, but we check for a `through` street. A street that changes class in between is probably not just a normal road but a different scenario